rclcpp_components/CMakeLists.txt

```diff
@@ -90,6 +90,25 @@ if(BUILD_TESTING)
   set(components "")
   add_library(test_component SHARED test/components/test_component.cpp)
   target_link_libraries(test_component PRIVATE component)
+  # Fake version of this macro so that rclcpp_components_register_node works
+  macro(_rclcpp_components_register_package_hook)
+    set(rclcpp_components_NODE_TEMPLATE ${CMAKE_CURRENT_SOURCE_DIR}/src/node_main.cpp.in)
+    add_library(rclcpp_components::component ALIAS component)
+  endmacro()
+  include(./cmake/rclcpp_components_register_node.cmake)
+  rclcpp_components_register_node(test_component
+    PLUGIN "test_rclcpp_components::DoesNotExist"
+    EXECUTABLE test_component_does_not_exist_executable
+  )
+  find_program(BASH_PROGRAM bash)
```

Comment:
Does this work as cross-platform? e.g Windows?

Reply:
Yeah it probably doesn't. Need to find a more portable way to do this.

Reply:
- [ ] make test work cross-platform

Reply:
Honestly, I'm going to suggest we don't bother adding a test for this.  While I'm OK with adding a check for this situation (though see below), it is not something a user can really run into if they are using our tools.  So I'll just suggest removing the test completely.

Reply:
I think that I encountered this situation originally due to cached files in an install directory, which colcon didn't overwrite for some reason, even though I used packages-up-to. So yes, it may be a very weird situation that results in this state, although once it gets there it is a very confusing and weird failure mode.

I'm ok removing the test. I mainly added it to verify that what I added actually worked.

- [ ] Remove test